python/ray/air/util/data_batch_conversion.py

```diff
@@ -93,7 +93,7 @@ def convert_pandas_to_batch_type(
     elif type == DataType.NUMPY:
         if len(data.columns) == 1:
             # If just a single column, return as a single numpy array.
-            return data.iloc[:, 0].to_numpy()
+            return data[data.columns[0]].to_numpy()
```

Comment:
This is just returning an rvalue, we're not mutating here, so `data.iloc[:, 0].to_numpy()` should be fine.